### PR TITLE
Backend: Token-Entity; Frontend: Registration and Login

### PR DESCRIPTION
**Pull request by annikavennemann**
Backend
- I generated a Token-Entity which is related to the User. 
- By every user-login a new token is generated for this exact user.

Frontend
- Post data via registration form
- Fetch Token within login-function

### CODE REVIEW

File: backend/src/Entity/Token.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    /**
+     * @ORM\Column(type="datetime")
+     */
+    private $validuntil;
+
+    /**
+     * @ORM\ManyToOne(targetEntity=User::class, inversedBy="tokens")
+     * @ORM\JoinColumn(nullable=false)
+     */
+    private $user;
```
Comment (web-devil) [inline backend/src/Entity/Token.php:34]: jap das sieht viel besser aus ;)

File: backend/src/Serializer/TokenSerializer.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace App\Serializer;
+
+use App\Entity\Token;
+
+class TokenSerializer {
+    public function serialize($token) {
+        $tokenArray = [
```
Comment (web-devil) [inline backend/src/Serializer/TokenSerializer.php:9]: kann man ein eine zeile packen

\json_encode([....])
also ohne variable

File: backend/src/Serializer/UserSerializer.php [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                    'firstname' => $user->getFirstname(),
+                    'lastname' => $user->getLastname(),
+                    'email' => $user->getEmail(),
+                    'password' => $user->getPassword(),
+                    'startdate' => $user->getStartdate()
+                ];
+            }
+        } else {
+            $userJson[] = [
+                'firstname' => $users->getFirstname(),
```
Comment (web-devil) [inline backend/src/Serializer/UserSerializer.php:22]: duplicate => private method für user array

File: backend/src/Repository/ChecklistRepository.php [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        return $this->createQueryBuilder('c')
+            ->andWhere('c.exampleField = :val')
+            ->setParameter('val', $value)
+            ->getQuery()
+            ->getOneOrNullResult()
+        ;
+    }
+    */
+}
```
Comment (janabremer) [inline backend/src/Repository/ChecklistRepository.php:57]: I guess you can delete these automatically generated functions, if you do not use them. Same for your other repositories.

File: frontend/src/components/Login/Login.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
+import saveLocally from '../../lib/saveLocally';
+
+export default function Login() {
+
+    return (
+        <div className="App">
```
Comment (janabremer) [inline frontend/src/components/Login/Login.js:6]: Why do you use the classname "App" in your Login component?
Besides, if you use styled components, classnames will be generated automatically.

File: frontend/src/components/Registration/Form.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        firstName: '',
+        lastName: '',
+        email: '',
+        password: '',
+        startdate: '',
+    })
+    
+    return (
+        <form onSubmit={registerUser}>
+            <label htmlFor="firstname">
```
Comment (janabremer) [inline frontend/src/components/Registration/Form.js:14]: It should be enough if you either nest the input in the label or link them with the htmlFor attribute.

File: backend/src/Entity/User.php [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+            $token->setUser($this);
+        }
+
+        return $this;
+    }
+
+    public function removeToken(Token $token): self
+    {
+        if ($this->tokens->removeElement($token)) {
+            // set the owning side to null (unless already changed)
```
Comment (janabremer) [inline backend/src/Entity/User.php:144]: You might want to delete your comment here... ;)

**Review by janabremer — APPROVED** [review]
Looks good to me. Especially the token authentification! :) 
I just had some small comments.

File: backend/src/Controller/AuthenticationController.php [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+{
+    /**
+     * @Route("/login", methods={"POST"})
+     */
+    public function login(
+        Request $request,
+        UserRepository $userRepository,
+        TokenRepository $tokenRepository,
+        tokenSerializer $tokenSerializer,
+        UserSerializer $UserSerializer
```
Comment (luschnat-ziegler) [inline backend/src/Controller/AuthenticationController.php:26]: Do you need to inject the UserSerializer here? It is not used in the function as far as i can tell.

File: backend/src/Serializer/UserSerializer.php (hunk 3 above)
Comment (annikavennemann) [reply backend/src/Serializer/UserSerializer.php:22]: > private method für user array

makes sense, danke :)

File: backend/src/Repository/ChecklistRepository.php (hunk 4 above)
Comment (annikavennemann) [reply backend/src/Repository/ChecklistRepository.php:57]: true :)

File: frontend/src/components/Login/Login.js (hunk 5 above)
Comment (annikavennemann) [reply frontend/src/components/Login/Login.js:6]: That's just a leftover of the automatic react-setup. I'll delete this.

File: backend/src/Controller/AuthenticationController.php (hunk 8 above)
Comment (annikavennemann) [reply backend/src/Controller/AuthenticationController.php:26]: Yes you are right, thanks.

File: frontend/src/components/Registration/Form.js (hunk 6 above)
Comment (annikavennemann) [reply frontend/src/components/Registration/Form.js:14]: right, I changed that :)

**Pull request merged**